Tools/ardupilotwaf/mavgen.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+# encoding: utf-8
+# (c) Siddharth Bharat Purohit, 3DRobotics Inc.
+
+"""
+The **mavgen.py** program is a code generator which creates mavlink header files.
+"""
+
+from waflib import Task
+from waflib.TaskGen import extension
+import os
+
+class mavgen(Task.Task):
+	"""generate mavlink header files"""
+	color   = 'BLUE'
+	run_str = '${MAVGEN} --lang=C --wire-protocol=1.0 --output=generated_src/include/mavlink/v1.0/ ${SRC[0].abspath()}'
+
+@extension('.xml')
+def process_mavgen(self, node):
+	"""
+	Create a mavgen task
+	"""
+	tsk = self.create_task('mavgen', node, [])
+
+def configure(cfg):
+	"""
+	setup environment for mavlink header generator
+	"""
+	env = cfg.env
+	cfg.env.env = dict(os.environ)
+
+	cfg.start_msg('Checking for mavlink submodule')
+	env.MAVLINK_DIR = cfg.srcnode.find_dir('modules/mavlink/').abspath()
+	if not cfg.srcnode.find_dir('modules/mavlink/'):
+		cfg.end_msg('not initialized, please run: git submodule init and git submodule update', color='RED')
+		return
+	cfg.end_msg('success')
+ 	env.MAVLINK_HEADERS = cfg.srcnode.abspath()  + '/build/generated_src/include/mavlink/v1.0/'
+ 	
+ 	env.MAVGEN = env.MAVLINK_DIR  + '/pymavlink/tools/mavgen.py'
+ 	cfg.env.env['PYTHONPATH'] = env.MAVLINK_DIR
+	cfg.start_msg('Checking for message_definitions')
+ 	env.MAV_MSG_DEFS = cfg.srcnode.find_resource('modules/mavlink/message_definitions/v1.0/ardupilotmega.xml').abspath()
+	if not cfg.srcnode.find_resource('modules/mavlink/message_definitions/v1.0/ardupilotmega.xml'):
```

Comment:
take a look on the gbenchmark tool. We separate the node creation and setting the variables in `env`. You could use this strategy and check if there's the node before setting env.